### PR TITLE
Bartending+: Shaking and Stirring

### PR DESCRIPTION
**Pull request by Vermidia**
## About the PR

Added two new reaction types: shaking and stirring
Shaking is done with the shaker using the shake action, while stirring is done with any type of spoon.

Changed a few drinks to use these new mixing types.

## Why / Balance

Adds a bit more variety to bartending gameplay using the tools they start with (and makes the shaker more than just a big container with a shake action).

## Technical details

ShakableSystem now will try to get the ReactionMixerComponent and mix chemicals through that if possible. Added a MixOnInteract bool to the ReactionMixerComponent so you can click things with the shake to mix them.

Added a doafterevent for reactionmixer and a TimeSpan in the component so using mixers can take time.

## Media

- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

https://github.com/space-wizards/space-station-14/assets/32827189/5f5ba48b-b8ce-4792-a576-6569ea25a8b8

## Breaking changes

nope
**Changelog**

:cl:
- add: Some drink reactions now require shaking with the shaker or stirring with a spoon

### CODE REVIEW

**Comment by ps3moira** [comment]
beautiful bartenders bartending

**Comment by beck-thompson** [comment]
Love it! Really good idea that brings some more interesting gameplay to bartenders 👍

**Comment by mirrorcult** [comment]
good changes

File: Content.Shared/Nutrition/EntitySystems/ShakeableSystem.cs [hunk 1]
```diff
@@ -1,3 +1,5 @@
+using Content.Shared.Chemistry.EntitySystems;
+using Content.Shared.Chemistry.Reaction;
```
Comment (Tayrtahn) [inline Content.Shared/Nutrition/EntitySystems/ShakeableSystem.cs:2]: These look like leftovers from an earlier version and are safe to delete.

File: Content.Shared/Nutrition/EntitySystems/ShakeableSystem.cs [hunk 2]
```diff
@@ -15,6 +17,7 @@ public sealed partial class ShakeableSystem : EntitySystem
     [Dependency] private readonly SharedAudioSystem _audio = default!;
     [Dependency] private readonly SharedDoAfterSystem _doAfter = default!;
     [Dependency] private readonly SharedHandsSystem _hands = default!;
+    [Dependency] private readonly SharedSolutionContainerSystem _solution = default!;
```
Comment (Tayrtahn) [inline Content.Shared/Nutrition/EntitySystems/ShakeableSystem.cs:20]: This too.

File: Resources/Prototypes/Recipes/Reactions/drinks.yml [hunk 3]
```diff
@@ -76,6 +82,8 @@
 
 - type: reaction
   id: B52
+  requiredMixerCategories:
+  - Shake
```
Comment (Tayrtahn) [inline Resources/Prototypes/Recipes/Reactions/drinks.yml:86]: The B52 is a layered drink, so don't think shaking is appropriate.

7428618800b0433caae86bb33e3c98f5

**Review by Tayrtahn — APPROVED** [review]

**Comment by NotSoDana** [comment]
Love you, man

**Comment by Terraspark4941** [comment]
not a 'tender, but really cool

**Pull request merged**